### DISCUSSION TITLE
Parse formated json from file

### DISCUSSION THREAD

**Discussion by hka** [nlohmann/json#3364] [Q&A] [answered]
I am trying to write a function that takes a file containing json and parsing it.
The following works for if I write unformatted json to a file (`std::fstream out; nlohmann::json json; out << json;`)
```
  using json = nlohmann::json;
  std::fstream in(path, std::fstream::in);
  if(!in.is_open())
  {
    return;
  }
  std::string line;
  while(std::getline(in, line))
  {
    json j= json::parse(line);
    //do stuff with j
  }
```
but if I have formatted json (`std::fstream out; nlohmann::json json; out << std::setw(4)<< json;`)
then it fails (as get line doesn't get a complete {} pair.

If I do 
`json j; in >> j;`
it works fine until it tries to parse EOF into the json object. What is the recommended way to parse something like:
```
{
    "name":"John",
    "age":0
}
{
    "name":"Doe",
    "age":0
}
```

**Comment by nlohmann** [comment, marked as answer]
This input is not valid JSON. It is similar to JSON Lines which is described here: https://json.nlohmann.me/features/parsing/json_lines/

**Reply by hka** [reply]
Ah right of course. I just looked at the [type conversion example](https://github.com/nlohmann/json#arbitrary-types-conversions), put a loop around it on the output side and then tried to do the same on the input side without thinking. Now I see why it doesn't work :) Thanks!
